src/main/java/org/folio/services/budget/CreateBudgetService.java

```diff
@@ -120,13 +122,38 @@ private Future<SharedBudget> createPlannedBudget(SharedBudget sharedBudget, Requ
     return createNewBudget(sharedBudget, requestContext);
   }
 
-  public Future<SharedBudget> createNewBudget(SharedBudget sharedBudget, RequestContext requestContext) {
+  private Future<SharedBudget> createNewBudget(SharedBudget sharedBudget, RequestContext requestContext) {
     double allocatedValue = sharedBudget.getAllocated();
     sharedBudget.setAllocated(0d);
-    return restClient.post(resourcesPath(BUDGETS_STORAGE), BudgetUtils.convertToBudget(sharedBudget), Budget.class, requestContext)
+    SharedBudget.BudgetStatus desiredStatus = sharedBudget.getBudgetStatus();
+    if (sharedBudget.getBudgetStatus() != SharedBudget.BudgetStatus.ACTIVE && allocatedValue != 0d) {
+      sharedBudget.setBudgetStatus(SharedBudget.BudgetStatus.ACTIVE);
+    }
+    return createActiveBudgetWithAllocation(BudgetUtils.convertToBudget(sharedBudget), allocatedValue, requestContext)
+      .compose(budget -> updateBudgetStatus(budget, Budget.BudgetStatus.fromValue(desiredStatus.value()), requestContext))
```

Comment:
in user story is written:
When status is “Inactive”, “Closed” or “Frozen” and positive value is presented we should automatically change status to Active on BE side.
If there is positive allocation and no changes to allocation value - user can change the status.

Interesting what it means, the second sentence means that we need to use original status at the end, in our case 'Inactive"?

Reply:
Joe didn't know if it would be possible to allocate money to an inactive fund, which is why he initially thought of changing the status to Active. But it is possible to make the budget active, create it, allocate money, and then change the status to what was desired if it was not Active. This is what is implemented here. You can take a look at the comments for details.

Reply:
ok please update AC criterias in this case or ask Joe to update, because QAs use AC for test cases creating